File: py/p2.py
#!/usr/bin/env python
#
# Finds the sum of the even-valued terms of the Fibonacci sequence not
# exceeding four million.
#
# (c) 2013 Matthew Rheaume

def solve():
	a = 0
	b = 1
	total = 0

	while (b < 4000000):
		x = a + b
		if (x % 2 == 0):
			total += x

		a = b
		b = x
	
	return total

if __name__ == "__main__":
	print solve()

File: hs/p2.hs
-- Finds the sum of the even-valued terms of the Fibonacci sequence not
-- exceeding four million.
--
-- (c) 2013 Matthew Rheaume

module Main where

import Data.Bits

fibSumEven :: Int -> Int -> Int
fibSumEven a b
	| x >= 4000000 = 0
	| testBit x 0 = fibSumEven b x
	| otherwise = x + fibSumEven b x
	where x = a + b

-- Sequence starts with 1 and 2
solve :: Int
solve = 2 + (fibSumEven 1 2)

main :: IO ()
main = putStrLn $ show $ solve
